src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -563,6 +562,30 @@ private Symbol ResolveMultipleSymbolsInAttributeTypeLookup(ArrayBuilder<Symbol>
             return null;
         }
 
+        private static bool NameAndArityMatchRecursively(Symbol x, Symbol y)
+        {
+            while (true)
+            {
+                if (isRoot(x))
+                {
+                    return isRoot(y);
+                }
+                if (isRoot(y))
+                {
+                    return false;
+                }
+                if (x.Name != y.Name || mangleName(x) != mangleName(y) || x.GetArity() != y.GetArity())
```

Comment:
>mangleName(x) != mangleName(y) [](start = 40, length = 30)

Why would this matter for the purpose of the name lookup? Mangling is not a language concept and Name is stripped of it. Do we have a test demonstrating significance of this check?

Reply:
Also, is this really consistent with SymbolDisplayFormat.QualifiedNameArityFormat?

---
In reply to: [405714806](https://github.com/dotnet/roslyn/pull/43181#discussion_r405714806) [](ancestors = 405714806)

Reply:
Yes, this is for consistency with `QualifiedNameArityFormat`. See `PropertyTests.CanNotReadPropertyFromAmbiguousGenericClass()` for a test that fails without this.

---
In reply to: [405715480](https://github.com/dotnet/roslyn/pull/43181#discussion_r405715480) [](ancestors = 405715480,405714806)

Reply:
> See PropertyTests.CanNotReadPropertyFromAmbiguousGenericClass() for a test that fails without this.

How exactly does it fail?

---
In reply to: [405765060](https://github.com/dotnet/roslyn/pull/43181#discussion_r405765060) [](ancestors = 405765060,405715480,405714806)

Reply:
The following error is reported:
```
(4,16): error CS0433: The type 'A<T>' exists in both
    '_0_09f9df97-a228-4ca4-9b71-151909f205e6, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' and
    '_0_09f9df97-a228-4ca4-9b71-151909f205e6, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'
```
instead of:
```
(4,16): error CS0104: 'A<>' is an ambiguous reference between 'A<T>' and 'A<T>'
```

---
In reply to: [405767894](https://github.com/dotnet/roslyn/pull/43181#discussion_r405767894) [](ancestors = 405767894,405765060,405715480,405714806)

Reply:
I think we should accept the baseline change. I think the original behavior (pay attention to mangling) was not intentional and from the language point of view both types have the same name.

---
In reply to: [405774516](https://github.com/dotnet/roslyn/pull/43181#discussion_r405774516) [](ancestors = 405774516,405767894,405765060,405715480,405714806)